File: Ejercicio7.py
# Evaluación de los empleados - Python

def evaluar_puntuacion(puntuacion):
    """
    Recibe una puntuación (0-100) y devuelve una tupla (nivel, recompensa).
    Niveles / Recompensas por defecto:
      90-100 : "Excelente" -> 1000.0
      75-89  : "Muy bueno"  ->  500.0
      60-74  : "Aceptable"  ->  200.0
      0-59   : "Necesita mejorar" -> 0.0
    Lanza ValueError si la puntuación está fuera de 0..100.
    """
    if puntuacion < 0 or puntuacion > 100:
        raise ValueError("La puntuación debe estar entre 0 y 100.")
    if puntuacion >= 90:
        return ("Excelente", 1000.0)
    elif puntuacion >= 75:
        return ("Muy bueno", 500.0)
    elif puntuacion >= 60:
        return ("Aceptable", 200.0)
    else:
        return ("Necesita mejorar", 0.0)

def main():
    try:
        entrada = input("Ingresa la puntuación del empleado (0-100): ").strip()
        score = float(entrada)
    except ValueError:
        print("Entrada inválida: por favor ingresa un número entre 0 y 100.")
        return

    try:
        nivel, recompensa = evaluar_puntuacion(score)
    except ValueError as e:
        print("Error:", e)
        return

    print(f"Nivel de rendimiento: {nivel}")
    print(f"Recompensa monetaria: ${recompensa:.2f}")

if __name__ == "__main__":
    main()

File: Ejercicio7.hs
-- Evaluación de los empleados - Haskell

evaluarEmpleado :: Double -> (String, Double)
evaluarEmpleado p
  | p < 0 || p > 100 = error "La puntuacion debe estar entre 0 y 100."
  | p >= 90          = ("Excelente", 1000.0)
  | p >= 75          = ("Muy bueno", 500.0)
  | p >= 60          = ("Aceptable", 200.0)
  | otherwise        = ("Necesita mejorar", 0.0)

-- Helper para mostrar el resultado sin romper en tiempo de ejecución
mostrarEvaluacion :: Double -> String
mostrarEvaluacion p
  | p < 0 || p > 100 = "Entrada inválida: la puntuación debe estar entre 0 y 100."
  | otherwise =
      let (nivel, recompensa) = evaluarEmpleado p
      in "Nivel de rendimiento: " ++ nivel ++ "\nRecompensa monetaria: $" ++ show recompensa

-- Ejemplo de main para consola
main :: IO ()
main = do
  putStrLn "Ingresa la puntuación del empleado (0-100):"
  linea <- getLine
  case reads linea :: [(Double, String)] of
    [(p, "")] -> putStrLn (mostrarEvaluacion p)
    _         -> putStrLn "Entrada inválida: por favor ingresa un número entre 0 y 100."